include/runtime/collections/RBTree.h

```diff
@@ -6,17 +6,18 @@
  * https://matt.might.net/papers/germane2014deletion.pdf                      */
 
 #include <cassert>
+#include <fmt/format.h>
 #include <functional>
 #include <memory>
 #include <stdexcept>
 #include <string>
 #include <utility>
 
-#define CONSTRUCT_MSG_AND_THROW(msg)                                           \
+#define CONSTRUCT_MSG_AND_THROW(...)                                           \
   do {                                                                         \
-    std::string emsg = std::string("[") + std::string(__func__)                \
-                       + std::string("] ") + std::string(msg);                 \
-    throw std::invalid_argument(emsg);                                         \
+    auto err_msg                                                               \
+        = ::fmt::format("[{}]: {}", __func__, ::fmt::format(__VA_ARGS__));     \
+    throw std::invalid_argument(err_msg);                                      \
   } while (false)
```

Comment:
This code is duplicated here: https://github.com/runtimeverification/llvm-backend/blob/fd2f7cc89cee542ba873c7f73f100a277e948d17/include/runtime/header.h#L282-L287

Reply:
I realize that. The issue is that I intentionally avoided including `runtime/header.h` and use the `KLLVM_HOOK_INVALID_ARGUMENT` directly. The reason is that I intend to include the header file `rangemap.h` (which includes `RBTree.h`) in `include/runtime/header.h` when the hooks PR is merged.

Would it be possible to either leave this as is, or extract the `KLLVM_HOOK_INVALID_ARGUMENT` and `fmt` dependency in a separate (e.g., `fmt_error_printing.h`) header file in `include/runtime`, and including that in both `include/runtime/header.h` and my header file as part of a separate PR?

Reply:
My preference would be to extract the formatting code to a separate header included by both the datastructure and `header.h`

Reply:
Addressed.